muonDIS/add_muonresponse.py

```diff
@@ -0,0 +1,71 @@
+import ROOT as r
+import os
+import argparse
+
+parser = argparse.ArgumentParser(description='Script to collect muons hitting the SBT')
+parser.add_argument('-f', '--inputfile', default='ship.conical.muonDIS-TGeant4.root')
+parser.add_argument('-m', '--muonfile', default='muonDis_0.root')
+args = parser.parse_args()
+
+# Open the existing ROOT file in update mode
+inputFile = r.TFile.Open(args.inputfile, "UPDATE") 
+sTree = inputFile.Get("cbmsim")  
+
+# Open the external file with additional vetoPoints
+muonFile = r.TFile.Open(args.muonfile, "READ")  
+muonTree = muonFile.Get('DIS') 
+
+# Create a new file for the updated tree
+outputFile = r.TFile('ship.conical.muonDIS-TGeant4_temp.root', "RECREATE")
+newTree = sTree.CloneTree(0)  # Clone the structure of the existing tree, but do not copy the entries
+
+# Access the vetoPoints branch in the original and external trees
+originalVetoPoints  = r.TClonesArray("vetoPoint")
+muonVetoPoints      = r.TClonesArray('vetoPoint')
+combinedVetoPoints  = r.TClonesArray("vetoPoint")
+
+sTree.SetBranchAddress("vetoPoint", originalVetoPoints)
+muonTree.SetBranchAddress("muon_vetoPoints", muonVetoPoints)
+newTree.SetBranchAddress("vetoPoint", combinedVetoPoints )
+
+# Loop over all entries and combine the vetoPoints
+for i in range(sTree.GetEntries()):
+    
+    sTree.GetEntry(i)
+    muonTree.GetEntry(i)
+
+    interaction_point = r.TVector3()  
+    sTree.MCTrack[0].GetStartVertex(interaction_point)         
+
+
+    # Clear and update the vetoPoints
+    combinedVetoPoints.Clear()
+    
+    index=0
+
+    # Copy original vetoPoints
+    for hit in originalVetoPoints:
+        if combinedVetoPoints.GetSize() == index:
+            combinedVetoPoints.Expand(index+1)
+        combinedVetoPoints[index] = hit
+        index+=1
+    
+    for hit in muonVetoPoints:
+        if hit.GetZ()<interaction_point.Z():
+            if combinedVetoPoints.GetSize() == index:
+                combinedVetoPoints.Expand(index+1)
+            combinedVetoPoints[index] = hit
+            index+=1
+            
+    newTree.Fill()
+
+# Write the updated tree to the output file
+newTree.Write()
+outputFile.Close()
+inputFile.Close()
+muonFile.Close()
+
+# Replace the old file with the new file
+os.replace('ship.conical.muonDIS-TGeant4_temp.root', args.inputfile)
```

Comment:
I'll open it up for reading in this case then.

Reply:
I don't quite remember, but I think I meant that we might as well just update the input file, at least if it's local instead of writing a new file and overwriting.

Reply:
We might however have to be careful with files on EOS, updating them in place might not be ideal. Maybe we can add a warning/error to tell the user to perform the update locally?

Reply:
yes, I'm also wondering of ways to block rewrite if it is already added (this will reapply the muon hits on the file if rerun). Perhaps a new naming? But this will break the default input file name for any existing analysis scripts.

Reply:
Could you check for the presence of a branch?
Alternatively, we could add some information to the file that could be checked, e.g. to the TaskList

Reply:
> We might however have to be careful with files on EOS, updating them in place might not be ideal. Maybe we can add a warning/error to tell the user to perform the update locally?

Since these are only intended for muonDIS files, should'nt this be allowed to edit in EOS?